libsolidity/ast/ASTAnnotations.h

```diff
@@ -128,6 +128,12 @@ struct TypeDeclarationAnnotation: DeclarationAnnotation
 	std::string canonicalName;
 };
 
+struct StructDeclarationAnnotation: TypeDeclarationAnnotation
+{
+	/// Whether the struct is recursive. Will be filled in by the DeclarationTypeChecker.
```

Comment:
Please clarify what exactly recursion means. Does it only concern direct recursion or also recursion via dynamically-sized arrays?